src/utils.ts

```diff
@@ -1142,44 +1142,55 @@ export class BufferPool {
 
 /** @public */
 export class HostAddress {
-  host;
-  port;
+  host: string | undefined;
+  port: number | undefined;
   // Driver only works with unix socket path to connect
   // SDAM operates only on tcp addresses
-  socketPath;
-  isIPv6;
+  socketPath: string | undefined;
+  isIPv6 = false;
 
   constructor(hostString: string) {
     const escapedHost = hostString.split(' ').join('%20'); // escape spaces, for socket path hosts
-    const { hostname, port } = new URL(`mongodb://${escapedHost}`);
 
     if (escapedHost.endsWith('.sock')) {
       // heuristically determine if we're working with a domain socket
       this.socketPath = decodeURIComponent(escapedHost);
-    } else if (typeof hostname === 'string') {
-      this.isIPv6 = false;
+      delete this.port;
+      delete this.host;
```

Comment:
I should've called this out, I'm being overly defensive here, newer `"target"` settings of typescript will make these properties exist because it'll emit JS that support class property syntax, and we have assertions that say these properties should not exist. This is not necessary now though, and/or we could also relax the tests.

Reply:
Interesting! I knew that this was happening for initialized class properties, but not uninitialized ones.

In any case, I'll resolve this, but it's interesting because `delete obj.property` has traditionally been (and I think still is) a performance cliff in V8 because it changes the shape of the object after the fact.

Reply:
https://www.typescriptlang.org/tsconfig#useDefineForClassFields
So it will start with ES2022 unless you flip this config. Perf cliff sounds like something worth fixing esp since this constructor is called quite frequently, I will adjust 😄